x-pack/solutions/observability/plugins/observability_ai_assistant_app/scripts/evaluation/kibana_client.ts

```diff
@@ -124,13 +124,13 @@ export class KibanaClient {
     return this.axios<T>({
       method,
       url,
-      data: data || {},
+      ...(method.toLowerCase() !== 'delete' ? { data: data || {} } : {}),
```

Comment:
why is this needed?

Reply:
Without this condition, deleting ruleIds fails here - https://github.com/elastic/kibana/pull/204574/files#diff-23cc9139c91a064a3ca574552ad823023c579cc2c68ff7f277c392102a0d526aL139

Because the `DELETE` method doesn't allow an `undefined` or empty body.

<img width="624" alt="Screenshot 2024-12-24 at 8 38 09 AM" src="https://github.com/user-attachments/assets/fc984da6-5007-4975-940f-7620381e2eae" />

Reply:
Can we change this to checking whether `data` is empty and if so, not setting the key? IIRC there actually are some routes in Kibana that allow for a request body with the DELETE method (whether that's a good idea or not 😄 )

Reply:
Sure, will do.